lib/src/repo_path.rs

```diff
@@ -83,6 +83,21 @@ impl RepoPath {
         }
     }
 
+    /// Converts repo-relative `Path` to `RepoPath`.
+    ///
+    /// The input path should not contain `.` or `..`.
+    pub fn from_relative_path(relative_path: impl AsRef<Path>) -> Option<Self> {
```

Comment:
Maybe this should return an `Error` directly?

Reply:
I considered that, but `parse_fs_path()` should include the pre-normalized (cwd-relative) `input` path in the error message. Perhaps, we can refactor the error type to not attach the `input: PathBuf`, and let the caller to do that.

Reply:
That sounds reasonable, but it might be still worth changing this to return `Result<Self, FsPathParseError>` with a new, different case (not containing the `PathBuf`), since the `None` return value here seems to logically represent an error. Up to your discretion.

Reply:
Split this to new PR: https://github.com/martinvonz/jj/pull/2603
Can you take a look?

Reply:
I'm going to merge this as is. I'll revisit the error stuff later.